needle/modules/dynamic/watch/traffic_capture.py

```diff
@@ -0,0 +1,163 @@
+                        When DNS poisoning results uneffective (applications communicating directly using IP addresses), this module enables\
+                        traffic to be intercepted regardlessly.',
+        'options': (
+            ('proxy_port', '8080', True,
+             'Port of the service that will handle the captured traffic'),
+            ('device_port', '9999', True,
+             'Loopback port on the device used for remote forwarding'),
+            ('outbound_ports', '80,443', True,
+             'List of outbuond ports the module has to capture the traffic from. Syntax: <port>[,<port>]*'),
+        ),
+    }
+
+    # ==================================================================================================================
+    # UTILS
+    # ==================================================================================================================
+    def module_pre(self):
+        return BackgroundModule.module_pre(self, bypass_app=True)
+
+    def _parse_ports(self, ports):
+        ports_list = ports.split(",")
+
+        ports_string = "{"
+        for port in ports_list:
+            ports_string += str(int(port)) + ","
+        ports_string = ports_string[:-1]
+        ports_string += "}"
+
+        return ports_string
+
+    # ==================================================================================================================
+    # REMOTE PORT FORWARDING
+    # ==================================================================================================================
+
+    def _handler(self, chan, host, port):
+        sock = socket.socket()
+        try:
+            sock.connect((host, port))
+        except Exception as e:
+            self.printer.error(
+                'Forwarding request to %s:%d failed: %r' % (host, port, e))
+            return
+
+        while True:
+            r, w, x = select.select([sock, chan], [], [])
+            if sock in r:
+                data = sock.recv(1024)
+                if len(data) == 0:
+                    break
+                chan.send(data)
+            if chan in r:
+                data = chan.recv(1024)
+                if len(data) == 0:
+                    break
+                sock.send(data)
+        chan.close()
+        sock.close()
+
+    def _reverse_forward_tunnel(self, server_port, remote_host, remote_port, transport):
+        transport.request_port_forward('', server_port)
+        while True:
+            chan = transport.accept(1000)
+            if chan is None:
+                continue
+            # Directing traffic captured from server_port to remote_port on the
+            # workstation
+            self._handler(chan, remote_host, remote_port)
+
+    def _remote_portforward_start(self):
+
+        localhost = "127.0.0.1"
+
+        # Create new ssh connection
+        client = paramiko.SSHClient()
```

Comment:
please refer to https://github.com/mwrlabs/needle/blob/master/needle/core/device/device.py#L81 to see how to instaurate an SSH connection